src/common/Chooser.cpp

```diff
@@ -21,28 +21,28 @@ using namespace CHOOSER;
 
 namespace
 {
-IRepresentationChooser* GetReprChooser(std::string_view type)
+std::unique_ptr<IRepresentationChooser> GetReprChooser(std::string_view type)
```

Comment:
another sidenote: you have changed the return value (here and other places) to `unique_ptr`
but in all places when you use a unique_ptr you should use `std::move`
because you must transfer ownership of the ptr, this is missing everywhere